lib/eslint/eslint-helpers.js

```diff
@@ -421,20 +421,12 @@ async function globMultiSearch({ searches, configLoader, errorOnUnmatchedPattern
         )
     );
 
-    const filePaths = [];
-
     for (let i = 0; i < results.length; i++) {
```

Comment:
Can we add a comment here explaining what we're looping over `results` twice? Especially without a test to validate this change, having some comments here would be very helpful for long-term maintenance.